### PR TITLE
DaemonSet

### PR DESCRIPTION
**Pull request by dmrogers7**
- Add DaemonSet status checks
- Add Spinnaker labels to DaemonSets
- Add Spinnaker annotations to DaemonSets

### CODE REVIEW

**Comment by dmrogers7** [comment]
I believe that there is a duplicate check in clouddriver's  [KubernetesDaemonSetHandler](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesDaemonSetHandler.java) class

This check
```java
  if (!generationMatches(daemonSet, status)) {
    return Status.defaultStatus().unstable(UnstableReason.OLD_GENERATION.getMessage());
  }

  private boolean generationMatches(V1beta2DaemonSet daemonSet, V1beta2DaemonSetStatus status) {
    Optional<Long> metadataGeneration =
        Optional.ofNullable(daemonSet.getMetadata()).map(V1ObjectMeta::getGeneration);
    Optional<Long> statusGeneration = Optional.ofNullable(status.getObservedGeneration());

    return statusGeneration.isPresent() && statusGeneration.equals(metadataGeneration);
  }
```

Is the same as this one

```java
  Long observedGeneration = status.getObservedGeneration();
  if (observedGeneration != null
      && !observedGeneration.equals(daemonSet.getMetadata().getGeneration())) {
    return Status.defaultStatus().unstable("Waiting for daemonset spec update to be observed");
  }
```

There's no test in [KubernetesDaemonSetHandlerTest](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/op/handler/KubernetesDaemonSetHandlerTest.java) for the second check, so I think it's dead code, which is why I didn't include it.

File: pkg/kubernetes/annotation.go [hunk 1]
```diff
@@ -65,6 +65,22 @@ func (c *controller) AddSpinnakerAnnotations(u *unstructured.Unstructured, appli
 		}
 	}
 
+	if strings.EqualFold(gvk.Kind, "daemonset") {
+		ds := NewReplicaSet(u.Object)
+
+		// Add spinnaker annotations to the replicaset pod template.
```
Comment (billiford) [inline pkg/kubernetes/annotation.go:71]: Change this comment to reference "daemonSet" instead of "replicaset".

**Review by billiford — CHANGES_REQUESTED** [review]
Change one comment and we're good.

File: pkg/kubernetes/annotation.go (hunk 1 above)
Comment (dmrogers7) [reply pkg/kubernetes/annotation.go]: Line 69 looks pretty important to change as well

**Pull request merged**